lib/beaker/command_factory.rb

```diff
@@ -16,7 +16,10 @@ def assertions
     end
 
     def execute(command, options={}, &block)
-      result = self.exec(Command.new(command), options)
+      options[:prepend_cmds] ? 
```

Comment:
Ok I think we have an issue here.  Anything that is passed into Command.new inside options will be added after the command resulting in undesired behavior.  In my case if I simply pass options through without processing it contains {:pty=>true, :accept_all_exit_codes=>true} in addition to :prepend_cmds=>"cmds".  This results in a command like the following which are invalid:

<prepended_cmds> rpm -q puppet-enterprise --accept_all_exit_codes=true --pty=true.

This is why the initialize method is currently deleting the :cmdexe option so that it does not get added as a -- option to the end of any commands.  I was doing a similar thing and deleting the :prepend_cmds option but we clearly we can't delete all options that should not be appended :)

Not sure how we solve this without extending the execute method in command_factory.rb and Command initialize method in command.rb to take an additional prepend_commands argument instead. :(  Then the caller of execute or exec.(Command.new...) would have process options pull out :prepend_commands and pass that inside the prepend_commands argument.

Thoughts...?